package.json

```diff
@@ -19,7 +19,8 @@
     "react-dom": "^16.13.1",
     "react-flip-move": "^3.0.4",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "3.4.1"
+    "react-scripts": "3.4.1",
+    "shortid": "^2.2.16"
```

Comment:
before adding the quantity increment and decrement functionality to the cart I was adding the same product multiple times on the addToCart function. and because of that when I used product id to remove the product from the cart it removes multiple products with the same product id. so to make them unique shortid used to give each cart item added to cart unique id.